app/uk/gov/hmrc/breathingspaceifstub/model/Period.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.breathingspaceifstub.model
+
+import java.time.LocalDate
+import java.util.UUID
+
+import play.api.libs.json.Json
+
+final case class Period(periodId: UUID, startDate: LocalDate, endDate: LocalDate)
```

Comment:
Isn't endDate optional?

Reply:
If you check RequestPeriod you will see that endDate is optional. According to what Anthony said they always generate the endDate when missing in the Request.